lib/pengine/clone.c

```diff
@@ -697,9 +695,9 @@ clone_print(pe_resource_t * rsc, const char *pre_text, long options, void *print
             GList *list = g_hash_table_get_values(rsc->allowed_nodes);
 
             /* Custom stopped list for non-unique clones */
-            free(stopped_list);
-            stopped_list = NULL;
-            stopped_list_len = 0;
+            if (stopped_list != NULL) {
```

Comment:
I'm a little bit confused about the existing logic. If `!pcmk_is_set(rsc->flags, pe_rsc_unique` and `clone_data->clone_max > active_instances`, then we blow away the `stopped_list` without ever printing it. Then we may overwrite it with something else and print that.

Do we want to print the original `stopped_list` before we create the "Custom stopped list for non-unique clones"? If not, should we avoid building it?

Reply:
We don't want to print it, the point is that the default stopped list isn't appropriate for anonymous clones when more instances could run. Avoiding building it in the first place would make more sense.

However note that the *print() functions are deprecated and unused. The pcmk__output_t model obsoleted them, but they're implementations of public API, so we need to keep them around for now. For the most part we just ignore them, so really all of this isn't necessary. FYI the next API compatibility break is expected in mid-2024 (which would be the 2.2.0 or 3.0.0 release).

Reply:
That's a long way off. On the bright side, hopefully that's enough time to deprecate everything we want to deprecate before then.